### PR TITLE
Dev: Fixed issue : SQL error on update survey settings

### PR DESCRIPTION
**Pull request by Shnoulle**
Dev: Fixed issue : Update languageSettings when have it
Dev: unsure for ['surveyls_survey_id','required', 'on' => 'insert'], but think it's OK and better

### CODE REVIEW

File: protected/models/SurveyLanguageSetting.php [hunk 1]
```diff
@@ -92,6 +92,8 @@ public function relations()
     public function rules()
     {
         return [
+            ['surveyls_survey_id','required', 'on' => 'insert'],
+            ['surveyls_language','required', 'on' => 'insert'],
```
Comment (SamMousa) [inline protected/models/SurveyLanguageSetting.php]: You can use CRangeValidator instead:
['surveyls_language', 'in', 'range'=> array_keys($this->survey->allLanguages)]

File: protected/models/SurveyLanguageSetting.php [hunk 2]
```diff
@@ -92,6 +92,8 @@ public function relations()
     public function rules()
     {
         return [
+            ['surveyls_survey_id','required', 'on' => 'insert'],
```
Comment (SamMousa) [inline protected/models/SurveyLanguageSetting.php]: Use CExistValidator instead, since this is a foreign key.

Comment (Shnoulle) [inline protected/models/SurveyLanguageSetting.php]: Good idea :) insert only or not ?

File: protected/models/SurveyLanguageSetting.php (hunk 1 above)
Comment (Shnoulle) [inline protected/models/SurveyLanguageSetting.php]: Same than https://github.com/LimeSurvey/LimeSurvey/pull/373/files#r41513558 . Needed on insert sure, but needed after too .....

File: protected/models/SurveyLanguageSetting.php (hunk 2 above)
Comment (SamMousa) [inline protected/models/SurveyLanguageSetting.php]: I'd say add it for all scenarios, can always relax it later.

Comment (Shnoulle) [inline protected/models/SurveyLanguageSetting.php]: Yes, we always need it .... else we can remove it .... some borken system : must be fixed.

File: protected/models/SurveyLanguageSetting.php (hunk 1 above)
Comment (Shnoulle) [inline protected/models/SurveyLanguageSetting.php]: ... must be fixed : no ability to create SUrvey : $this->survey is null at create

Comment (SamMousa) [inline protected/models/SurveyLanguageSetting.php]: Rules are not checked at creation. Just set survey Id after construction
and it should work.

**Comment by Shnoulle** [comment]
Updated : no test on import done.

File: protected/models/SurveyLanguageSetting.php (hunk 1 above)
Comment (Shnoulle) [inline protected/models/SurveyLanguageSetting.php]: Rules are checked before survey is created : https://github.com/LimeSurvey/LimeSurvey/blob/develop/protected/controllers/SurveysController.php#L245 
2 solution : 
* we save survey BEFORE languagesettings (tested with : && $survey->save(false) && $languageSetting->validate() not working for language)
* we review languagesettings for 'on'=>'insert'

**Comment by Shnoulle** [comment]
https://github.com/LimeSurvey/LimeSurvey/commit/301c4e0ac6196239d5ce3c8585641b37a2cc1d20

**Pull request closed without merging**